src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const SEPARATOR = fromFormat[fromFormat.length - 1];
+  const JOINER = toFormat[toFormat.length - 1];
+  const DATEPARTS = date.split(SEPARATOR);
+
+  while (DATEPARTS.length > 1 && DATEPARTS[DATEPARTS.length - 1] === '') {
+    DATEPARTS.pop();
+  }
+
+  const DATEMAP = {};
+
+  fromFormat.forEach((part, index) => {
```

Comment:
The `forEach` loop is used to populate the `DATEMAP` object with date parts. However, there is no check to ensure that `DATEPARTS[index]` exists. This could lead to `undefined` values being assigned to `DATEMAP` properties if `DATEPARTS` has fewer elements than expected, which again highlights the need for input validation.